src/test/groovy/com/netflix/gradle/plugins/application/OspackageApplicationSpringBootPluginLauncherSpec.groovy

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2014-2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.gradle.plugins.application
+
+import com.netflix.gradle.plugins.deb.Scanner
+import nebula.test.IntegrationSpec
+
+import java.util.jar.Manifest
+import java.util.zip.ZipFile
+
+class OspackageApplicationSpringBootPluginLauncherSpec extends IntegrationSpec {
+    def 'application shows up in deb'() {
+        writeHelloWorld('nebula.test')
+        buildFile << """
+            repositories {
+                mavenCentral()
+            }
+
+            ${applyPlugin(OspackageApplicationSpringBootPlugin)}
+
+            dependencies {
+                compile 'org.springframework.boot:spring-boot-starter:1.5.10.RELEASE'
+            }
+        """.stripIndent()
+
+        when:
+        runTasksSuccessfully('buildDeb')
+
+        then:
+        final archivePath = file("build/distributions/${moduleName}_unspecified_all.deb")
+        final scanner = new Scanner(archivePath, new File("${getProjectDir()}/build/tmp/extract"))
+
+        final startScript = "./opt/${moduleName}/bin/${moduleName}"
+        final moduleJarName = "./opt/${moduleName}/lib/${moduleName}.jar"
+
+        0755 == scanner.getEntry(startScript).mode
+
+        [
```

Comment:
Strange. If I change one of the file names, the test fails as expected:

```
Condition failed with Exception:

[ startScript, "./opt/${moduleName}/bin/${moduleName}.bataoeu", moduleJarName].each { scanner.getEntry("${it}").isFile() }
  |                    |                 |                      |              |
  |                    |                 |                      |              java.io.FileNotFoundException: ./opt/application-shows-up-in-deb/bin/application-shows-up-in-deb.bataoeu
  |                    |                 |                      ./opt/application-shows-up-in-deb/lib/application-shows-up-in-deb.jar
  |                    |                 application-shows-up-in-deb
  |                    application-shows-up-in-deb
  ./opt/application-shows-up-in-deb/bin/application-shows-up-in-deb

	at com.netflix.gradle.plugins.application.OspackageApplicationSpringBootPluginLauncherSpec.application shows up in deb(OspackageApplicationSpringBootPluginLauncherSpec.groovy:52)
Caused by: java.io.FileNotFoundException: ./opt/application-shows-up-in-deb/bin/application-shows-up-in-deb.bataoeu
```

But if I insert the code you did, the test still passes.

Reply:
In fact, inserting `[false].each {false}` also passes so I'm suspecting no iteration is happening for some reason for `[false]` but I'm not able to reproduce that in a simple Groovy shell script. Anyway, the test is actually verifying for correct behavior.

Reply:
I think your case fails not because the assertion is false but because you are getting an exception. if `scanner.getEntry("${it}").isFile()` would return just `false` the test would pass.

Reply:
Ah, I see what you're saying. And to confirm, I placed `false` in the middle of the block and the test still passed.